From: Gerald Schaefer <gerald.schaefer@de.ibm.com>
Subject: s390/pci: fix removal of nonexistent pci bus
Patch-mainline: v3.14-rc1
Git-commit: 704268925d32a0457202371a61580af76b94c53a
References: fate#315311, LTC#92765

Summary:     s390: add pci support
Description: Add pci support for System z.

Upstream-Description:

             s390/pci: fix removal of nonexistent pci bus

             If we remove a pci bus after receiving a hotplug notification we need
             to check if the bus is actually present (creation of the pci bus
             during an earlier notification may have been failed).

             Reviewed-by: Gerald Schaefer <gerald.schaefer@de.ibm.com>
             Signed-off-by: Sebastian Ott <sebott@linux.vnet.ibm.com>


Signed-off-by: Gerald Schaefer <gerald.schaefer@de.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 arch/s390/pci/pci_event.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/s390/pci/pci_event.c
+++ b/arch/s390/pci/pci_event.c
@@ -112,6 +112,8 @@ static void __zpci_event_availability(st
 		clp_rescan_pci_devices();
 		break;
 	case 0x0308: /* Standby -> Reserved */
+		if (!zdev)
+			break;
 		pci_stop_root_bus(zdev->bus);
 		pci_remove_root_bus(zdev->bus);
 		break;
